### ISSUE TITLE
Delete text not added if formset is not visible

### ISSUE THREAD

**Issue by elo80ka** [elo80ka/django-dynamic-formset#54]
_From [bentap...@gmail.com](https://code.google.com/u/100406203710526534392/) on January 11, 2013 19:45:03_

What steps will reproduce the problem? Do anything that hides the formset before the formset plugin is initialised (for example, put it in an inactive tab). What is the expected output? What do you see instead? When the formset is made visible I would expect the delete link to be available. Instead it is missing. What version of the product are you using? On what operating system? jQuery Formset 1.3-pre. Please provide any additional information below. This is the code ...

if (row.is(':visible')) {
    insertDeleteLink(row);
    applyExtraClasses(row, i);
}

.. which was added in r14 .

Was the if statement that wraps insertDeleteLink(row) introduced to fix a bug with something else?

Removing the if statement works for me.

_Original issue: http://code.google.com/p/django-dynamic-formset/issues/detail?id=54_

**Comment by pkaczynski** [comment]
Same here. Any news? Is this project still maintained?

**Comment by elo80ka** [comment]
Hi @pkaczynski...yes, it's been kinda quiet around here :smile: 
Could you try moving the call to `insertDeleteLink(...)` outside the if-block, as mentioned in the comment? Let me know if that works for you, and I'll make the fix and update.

Cheers.

**Comment by pkaczynski** [comment]
Yes, it's working ok with that fix. Do you remember why did you introduce this condition in the code?

**Comment by elo80ka** [comment]
Nah...can't remember, was quite a while back. I'll add some tests for this and commit the fix tonight.

**Comment by JarnoRFB** [comment]
I faced the same problem when removing all forms with the DeleteLink and then reloading the page. The code that caused the issue was 
```js
 // Check if we're under the minimum number of forms - not to display delete link at rendering
                if (!showDeleteLinks()){
                	row.find('a.' + delCssSelector).hide();
                }
```

the piece of 

```js
if (row.is(':visible')) {
    insertDeleteLink(row);
    applyExtraClasses(row, i);
}
```

is actually unproblematic in this case, since the at least one row will be visible after reload. I think the problem is coming from the fact that the hidden input field `TOTAL_FORMS` has the value 0 after reloading from a point where all form in the formset where deleted. 
The same problem occurs with the `add` button. When the maximum number of forms is added before reload, the `add` button vanishes. 
Is this a problem of the Django `{{ formset.management_form }}` that does not catch the correct values after reload?

**Comment by firstTimeCaller** [comment]
This problem is still live
If the formset is loaded into a hidden div, then the div is made visible. if (row.is(':visible')) returns false.
Is there a use case where adding the remove buttons to a hidden element will cause an issue? Can if (row.is(':visible')) be removed?

**Comment by rhaamo** [comment]
Still a problem yes.
I have three formset in bootstrap tabs, the first one default works, no link for the ones in the hidden tabs.

Removing the if seems to work, but no ideas of possible side effects.

**Comment by rob101** [comment]
Can confirm this issue is still open.

**Comment by RogueOverride** [comment]
This is still an issue. It's been years since this issue was reported, is this issue still getting fixed?

**Comment by MahmudulHassan5809** [comment]
I am also facing this problem .I am using bootstrap tab .How can i solve this issue? any help please.

**Comment by regoawt** [comment]
There are now many other tickets related to this bug: 

- https://github.com/elo80ka/django-dynamic-formset/pull/181
- https://github.com/elo80ka/django-dynamic-formset/issues/177
- https://github.com/elo80ka/django-dynamic-formset/issues/107
- https://github.com/elo80ka/django-dynamic-formset/issues/97

Could the PR please be merged by anyone with permissions? @rhaamo @elo80ka @pkaczynski @rob101 @JarnoRFB @luizanao
